presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroupManager.java

```diff
@@ -156,6 +181,19 @@ public void setConfigurationManager(String name, Map<String, String> properties)
         ResourceGroupConfigurationManager configurationManager = configurationManagerFactory.create(ImmutableMap.copyOf(properties), () -> memoryPoolManager);
         checkState(this.configurationManager.compareAndSet(null, configurationManager), "configurationManager already set");
 
+        if (configurationManager instanceof ReloadingConfigurationManager) {
+            ((ReloadingConfigurationManager) configurationManager).addConfigReloadListener((ignored) -> {
```

Comment:
I don't think this listener approach is going to work. By clearing the `groups` and `rootGroups` previous queries that were submitted to a resource group, "a", will be running in that old object, but new ones will go to a new copy of "a", this could lead to the group going over quota.

I'm pretty sure you don't need to make any changes to this class actually. I designed the `InternalResourceGroup` objects to be thread-safe and re-configurable, so the `FileResourceGroupConfigurationManager` can just retain a reference to them (see the docs for `ResourceGroupConfigurationManager` which explicitly allow this) and then reconfigure the groups whenever a change to the file is detected